server/alerts_smtp.py

```diff
@@ -0,0 +1,66 @@
+import king_phisher.plugins as plugin_opts
+import king_phisher.server.plugins as plugins
+import king_phisher.server.signals as signals
+from os import environ
+
+
+try:
+	from smtp2go import core as smtp2go
+except ImportError:
+	has_smtp2go = False
+else:
+	has_smtp2go = True
+
+EXAMPLE_CONFIG = """\
+  api_key: <api_key>
+"""
+
+
+class Plugin(plugins.ServerPlugin):
+	authors = ['Spencer McIntyre', 'Mike Stringer']
+	title = 'Campaign Alerts: via SMTP2Go'
+	description = """
+	Send campaign alerts via the SMTP2go lib. This requires that users specify
+    their email through the King Phisher client to subscribe to notifications.
+	"""
+	homepage = 'https://github.com/securestate/king-phisher-plugins'
+	options = [
+		plugin_opts.OptionString(
+			name='api_key',
+			description='SMTP2GO API Key'
+		),
+		plugin_opts.OptionString(
+			name='server_email',
+			description='Server email address to send notifications from'
+		)
+	]
+	req_min_version = '1.10.0b0'
+	req_packages = {
+		'smtp2go': has_smtp2go
+	}
+	def initialize(self):
+		signals.campaign_alert.connect(self.on_campaign_alert)
+		return True
+
+	def on_campaign_alert(self, table, alert_subscription, count):
+		user = alert_subscription.user
+		if not user.email:
+			self.logger.debug("user {0} has no email address specified, skipping SMTP alert".format(user.id))
+			return False
+		environ['SMTP2GO_API_KEY'] = self.config['api_key']
```

Comment:
Can you please add a comment in here referencing the upstream PR you made that will make this cleaner? If we update this in the future we'll know that we can use your changes and why it was done this way today.